### ISSUE TITLE
Text layer position changes after scaling the component

### ISSUE THREAD

**Issue by ljh-coder** [hrynko/vue-pdf-embed#185]
hi,I have a problem.
In the `vue-pdf-embed@2.0.2` version,The pdf file display becomes blurry, this does not occur in the `vue-pdf-embed@1.2.1` version, I tried to use the `scale` parameter to deal with this, although the pdf file becomes clear, but when using the text selection, I find that the position of the text layer has changed, what should I do with it.

`vue-pdf-embed@2.0.2` version:
```js
<template>
    <vue-pdf-embed :source="doc" :page="1" :scale="3" text-layer="true"></vue-pdf-embed>
</template>

<script setup>
import {reactive} from 'vue';
import VuePdfEmbed, { useVuePdfEmbed } from "vue-pdf-embed";

// essential styles
import 'vue-pdf-embed/dist/style/index.css'

// optional styles
import 'vue-pdf-embed/dist/style/annotationLayer.css'
import 'vue-pdf-embed/dist/style/textLayer.css'

const pdfState = reactive({
    pdfSource: {
        url: "https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/web/compressed.tracemonkey-pldi-09.pdf",
        cMapUrl: 'https://cdn.jsdelivr.net/npm/pdfjs-dist@3.11.174/cmaps/',
        cMapPacked: true,
    },
})

const { doc } = useVuePdfEmbed({
    source: pdfState.pdfSource,
});

</script>
```
![2](https://github.com/hrynko/vue-pdf-embed/assets/62947861/26e9593f-6b71-4cf6-80af-b33edffc6239)

`vue-pdf-embed@1.2.1` version:
```js
<template>
    <vue-pdf-embed :source="pdfState.pdfSource" :page="1" text-layer="true"></vue-pdf-embed>
</template>

<script setup>
import {reactive} from 'vue';
import VuePdfEmbed from "vue-pdf-embed";

const pdfState = reactive({
    pdfSource: {
        url: "https://raw.githubusercontent.com/mozilla/pdf.js/ba2edeae/web/compressed.tracemonkey-pldi-09.pdf",
        cMapUrl: 'https://cdn.jsdelivr.net/npm/pdfjs-dist@2.9.359/cmaps/',
        cMapPacked: true,
    },
})
</script>
```
![1](https://github.com/hrynko/vue-pdf-embed/assets/62947861/3556b224-2e80-42bb-b2a7-87b49d67b1ac)

**Comment by markusgeert** [comment]
I'm also facing this. The problem seems to lie in the following line:
https://github.com/hrynko/vue-pdf-embed/blob/f36296e1eb225f3cebd1c64d7b1985cc6895268a/src/VuePdfEmbed.vue#L262

which messes up the scaling of the text and annotation layers when the `scale` prop is set, causing the shift.

If the goal of the `scale` prop is to control the sharpness of the pdf, we can change the call to the [`renderPage`](https://github.com/hrynko/vue-pdf-embed/blob/f36296e1eb225f3cebd1c64d7b1985cc6895268a/src/VuePdfEmbed.vue#L283) function to include the scale instead, which seems to work fine:

```ts
await renderPage(page, viewport.clone({ scale: props.scale }), canvas)
```

I'm down to create the PR that implements this, but I guess it'll technically be a breaking change. @hrynko what do you think?

**Comment by hrynko** [comment]
Ok, I see the problem, but I haven't had time to look into it yet. Maybe this problem can be solved by using different scaling factors, but @markusgeert's PR is also welcome.

**Comment by markusgeert** [comment]
I'll create a PR.

On a side-note, it seems that Firefox uses a scaling factor of 2 when rendering PDF's. That might also be a worthwhile change to make?

**Comment by hrynko** [comment]
> it seems that Firefox uses a scaling factor of 2 when rendering PDF's

Do you mean Firefox's internal viewer?

**Comment by hrynko** [comment]
Closing, as fixed in #186

Thank you, @markusgeert
